src/common/NetworkManager.ts

```diff
@@ -69,19 +69,43 @@ export class NetworkManager extends EventEmitter {
   _client: CDPSession;
   _ignoreHTTPSErrors: boolean;
   _frameManager: FrameManager;
-  _requestIdToRequest = new Map<string, HTTPRequest>();
+
+  /*
+   * There are four possible orders of events:
+   *  A. `_onRequestWillBeSent`
+   *  B. `_onRequestWillBeSent`, `_onRequestPaused`
+   *  C. `_onRequestPaused`, `_onRequestWillBeSent`
```

Comment:
when does the C sequence happen? So A is when there is no interception and B is the normal interception flow?

Reply:
Yes, that is correct. Apparently B is with cache disabled, and C is with cache enabled, but this behavior is undocumented.

Reply:
This module was always anticipating both B and C but it previously failed for D.
D is trivially fixed by preserving the entry in `_requestIdToRequestWillBeSentEvent` until `_onLoadingFinished` (and adding a guard clause for redirects as they have the same `requestId`): https://github.com/puppeteer/puppeteer/pull/7060/commits/5f183e68e691acc7b8115f7c09345921214e4a72#r613342144